src/lightning/pytorch/core/saving.py

```diff
@@ -56,11 +56,17 @@ def _load_from_checkpoint(
     map_location: _MAP_LOCATION_TYPE = None,
     hparams_file: Optional[_PATH] = None,
     strict: Optional[bool] = None,
+    weights_only: Optional[bool] = None,
```

Comment:
Should we default to `weights_only=None` or `weights_only=True`? If we have no use for `weights_only=None`, we can simplify the type hint to `weights_only: bool = True`.